src/mlpack/methods/ann/layer/flatten_t_swish_impl.hpp

```diff
@@ -0,0 +1,80 @@
+/**
+ * @file methods/ann/layer/flatten_t_swish_impl.hpp
+ * @author Fawwaz Mayda
+ *
+ * Definition of Flatten T Swish layer first introduced in the acoustic model,
+ * Hock Hung Chieng, Noorhaniza Wahid, Pauline Ong, Sai Raj Kishore Perla,
+ * "Flatten-T Swish: a thresholded ReLU-Swish-like activation function for deep learning", 2018
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_FLATTEN_T_SWISH_IMPL_HPP
+#define MLPACK_METHODS_ANN_LAYER_FLATTEN_T_SWISH_IMPL_HPP
+
+// In case it hasn't yet been included.
+#include "flatten_t_swish.hpp"
+#include <mlpack/methods/ann/activation_functions/logistic_function.hpp>
+#include <mlpack/methods/ann/activation_functions/rectifier_function.hpp>
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+template<typename InputDataType, typename OutputDataType>
+FlattenTSwish<InputDataType, OutputDataType>::FlattenTSwish(
+    const double T) : t(T)
+{
+  // Nothing to do here.
+}
+
+template<typename InputDataType, typename OutputDataType>
+template<typename InputType, typename OutputType>
+void FlattenTSwish<InputDataType, OutputDataType>::Forward(
+    const InputType& input, OutputType& output)
+{
+  // Holder for Relu and Sigmoid values.
+  OutputDataType relu, sigmoid;
+  RectifierFunction::Fn(input,relu);
+  LogisticFunction::Fn(input,sigmoid);
+
+  // F(x) = relu(x) * sigmoid(x) + t.
+  output.set_size(arma::size(input));
+  output = relu % sigmoid + t;
+}
+
+template<typename InputDataType, typename OutputDataType>
+template<typename DataType>
+void FlattenTSwish<InputDataType, OutputDataType>::Backward(
+    const DataType& input, const DataType& gy, DataType& g)
+{
+  DataType derivate, sigmoid;
+  LogisticFunction::Fn(input,sigmoid);
+  derivate.set_size(arma::size(input));
+  for(size_t i = 0; i < input.n_elem; ++i) 
+  {
+    if (input(i) >= 0)
+    {
+      double f = input(i) * sigmoid(i);
```

Comment:
```suggestion
      const DataType::elem_type f = input(i) * sigmoid(i);
```
The output type could be `float` as well, so let's use `DataType::elem_type`.